sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py

```diff
@@ -406,6 +406,151 @@ def _upload_blob_options(  # pylint:disable=too-many-statements
             raise ValueError("Unsupported BlobType: {}".format(blob_type))
         return kwargs
 
+    def _upload_blob_from_url_options(self, source_url, **kwargs):
+        # type: (...) -> Dict[str, Any]
+        headers = kwargs.pop('headers', {})
+        tier = kwargs.pop('standard_blob_tier', None)
+        include_source_blob_properties = kwargs.pop('include_source_blob_properties', True)
+        timeout = kwargs.pop('timeout', None)
+        dest_mod_conditions = get_modify_conditions(kwargs)
+        blob_tags_string = serialize_blob_tags_header(kwargs.pop('tags', None))
+        content_settings = kwargs.pop('content_settings', None)
+        source_content_md5 = kwargs.pop('source_content_md5', None)
+        if content_settings:
+            kwargs['blob_http_headers'] = BlobHTTPHeaders(
+                blob_cache_control=content_settings.cache_control,
+                blob_content_type=content_settings.content_type,
+                blob_content_md5=None,
```

Comment:
The service team has informed us that this header is still not behaving as expected. Hence it will be set to None in the meantime.